### PR TITLE
Support for v3 array parameters

### PR DESCRIPTION
**Pull request by SebMourlhou**
Here is a proposal for issue #149

### CODE REVIEW

File: lib/JSON/Validator/OpenAPI/Mojolicious.pm [hunk 1]
```diff
@@ -191,16 +191,49 @@ sub _build_formats {
...
+  my $custom_re;
+
+  # support for v3 style / explode
+  if ($p->{style}) {
+    if ($p->{style} eq 'simple') {
+      $collection_format = 'csv';
+    }
+    elsif ($p->{style} eq 'label') {
+      $custom_re = qr{\.};
+      $collection_format = $p->{explode} ? 'custom' : 'csv' if $data =~ s/^$custom_re//;
```
Comment (jhthorsen) [inline lib/JSON/Validator/OpenAPI/Mojolicious.pm]: Could you please elaborate when $custom_re is useful below? Could you use length() instead? Could you also send me a link to the spec where this is explained?

**Comment by SebMourlhou** [comment]
See https://swagger.io/docs/specification/serialization/

$custom_re is useful for the styles "label" and "matrix". The regex is just a dot for "label" but with "matrix", it is more complicated (depends on the name of the parameter).
$data must begin with this regex (explode true or false).
If explode is true, the delimitor is this regex, that's why $collection_format becomes "custom".
If explode is false, $collection_format is simply "csv".

$custom_re is then used if $collection_format is "custom" :
https://github.com/SebMourlhou/mojolicious-plugin-openapi/blob/3993d2ce8094650e8113523ca75f6c048bd22896/lib/JSON/Validator/OpenAPI/Mojolicious.pm#L234-L235

File: lib/JSON/Validator/OpenAPI/Mojolicious.pm (hunk 1 above)
Comment (SebMourlhou) [reply lib/JSON/Validator/OpenAPI/Mojolicious.pm]: Have you seen my answer given one month ago ?

Comment (jhthorsen) [reply lib/JSON/Validator/OpenAPI/Mojolicious.pm]: Sorry, I missed out that `$custom_re` was used outside of these if-statements.

**Pull request merged**